### PR TITLE
Add comments and format Axis._get_coord_info

### PR DESCRIPTION
**Pull request by Illviljan**
## PR Summary
Explain Axis._get_coord_info better by adding comments and an (atleast to me) easier to read format. Should be no change in functionality.

Simplifies review of #19873 once merged. 

## PR Checklist

- [N/A] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [N/A] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/mpl_toolkits/mplot3d/axis3d.py [hunk 1]
```diff
@@ -178,23 +178,35 @@ def get_rotate_label(self, text):
...
+        # Project the bounds along the current position of the cube:
+        bounds = mins[0], maxs[0], mins[1], maxs[1], mins[2], maxs[2]
+        bounds_proj = self.axes.tunit_cube(bounds, self.axes.M)
+
+        # Determine which one of the parallel planes are higher up:
+        highs = np.zeros(3, dtype=bool)
+        for i in range(3):
+            mean_z0 = np.mean(bounds_proj[self._PLANES[2 * i], 2])
+            mean_z1 = np.mean(bounds_proj[self._PLANES[2 * i + 1], 2])
+            highs[i] = mean_z0 < mean_z1
```
Comment (jklymak) [inline lib/mpl_toolkits/mplot3d/axis3d.py:207]: Are we sure this gets tested?  I agree the previous code was very hard to follow, so I support this change if its identical.

File: lib/mpl_toolkits/mplot3d/axis3d.py [hunk 2]
```diff
@@ -178,23 +178,35 @@ def get_rotate_label(self, text):
...
-        maxs = maxs + deltas / 4.
-
-        vals = mins[0], maxs[0], mins[1], maxs[1], mins[2], maxs[2]
-        tc = self.axes.tunit_cube(vals, self.axes.M)
-        avgz = [tc[p1][2] + tc[p2][2] + tc[p3][2] + tc[p4][2]
-                for p1, p2, p3, p4 in self._PLANES]
-        highs = np.array([avgz[2*i] < avgz[2*i+1] for i in range(3)])
-
-        return mins, maxs, centers, deltas, tc, highs
+        mins, maxs = np.array(
```
Comment (jklymak) [inline lib/mpl_toolkits/mplot3d/axis3d.py:181]: We don't usually add the extra lines per-bracket like this.

File: lib/mpl_toolkits/mplot3d/axis3d.py (hunk 1 above)
Comment (Illviljan) [reply lib/mpl_toolkits/mplot3d/axis3d.py:207]: It's basically this without all the tricky for loops:
```python
        avgz = [tc[p1][2] + tc[p2][2] + tc[p3][2] + tc[p4][2]
                     for p1, p2, p3, p4 in self._PLANES]
        highs = np.array([avgz[2*i] < avgz[2*i+1] for i in range(3)])
```

The `highs` bool determines when to switch upside down or not so it pretty noticeable if it wouldn't have worked. 

You are correct though that it is not exactly the same values. `avgz` sums them and I used `mean` instead because that's what the name `avgz` seems to imply, it doesn't matter though if 

`(x0, x1, x2, x3) < (y0, y1, y2, y3)` 

is true then 

`(x0, x1, x2, x3)/4 < (y0, y1, y2, y3)/4` 

will also remain true.

I added some prints, but with np.sum instead for easy comparison:
```python
        avgz = [tc[p1][2] + tc[p2][2] + tc[p3][2] + tc[p4][2]
                     for p1, p2, p3, p4 in self._PLANES]
        highs = np.array([avgz[2 * i] < avgz[2 * i + 1] for i in range(3)])

        print(avgz)
        print("nextsdf")

        # Determine which one of the parallel planes are higher up:
        highs = np.zeros(3, dtype=bool)
        for i in range(3):
            mean_z0 = np.sum(bounds_proj[self._PLANES[2 * i], 2])
            mean_z1 = np.sum(bounds_proj[self._PLANES[2 * i + 1], 2])
            print(mean_z0, mean_z1)
            highs[i] = mean_z0 < mean_z1

Results:
[-3.922656332439214, -4.1027538546066875, -3.833147670696861, -4.19226251634904, -3.908760756651483, -4.1166494303944186]
nextsdf
-3.922656332439214 -4.1027538546066875
-3.833147670696861 -4.19226251634904
-3.908760756651483 -4.1166494303944186
```

I find comparing a mean value makes more sense than comparing the summed value. But np.sum is probably a little faster.

File: lib/mpl_toolkits/mplot3d/axis3d.py (hunk 2 above)
Comment (Illviljan) [reply lib/mpl_toolkits/mplot3d/axis3d.py:181]: This is just what `black` suggests which is triggered every time I save the file in my IDE. I can switch it back, but I feel like matplotlib is swimming against the current here. 

There are so many larger python projects (pandas, dask, xarray etc) that uses it now that you likely will waste more and more time with these kinds of code style discussions.

But matplotlib uses a lot of linear algebra and matrix calculations which I think `black` is frankly terrible at, so I can understand the reluctance of relying on it completely.

Comment (anntzer) [reply lib/mpl_toolkits/mplot3d/axis3d.py:181]: We don't use black, see https://github.com/matplotlib/matplotlib/issues/18796 (your point that black is particularly bad at math expressions actually summarizes our problem quite well :-)).

File: lib/mpl_toolkits/mplot3d/axis3d.py (hunk 1 above)
Comment (jklymak) [reply lib/mpl_toolkits/mplot3d/axis3d.py:205]: No yours is definitely easier to read, and I can't imagine it is noticeably slower.

**Review by jklymak — APPROVED** [review]

File: lib/mpl_toolkits/mplot3d/axis3d.py [hunk 3]
```diff
@@ -183,18 +183,28 @@ def _get_coord_info(self, renderer):
...
-        highs = np.array([avgz[2*i] < avgz[2*i+1] for i in range(3)])
-
-        return mins, maxs, centers, deltas, tc, highs
+
+        # Get the mean value for each bound:
+        centers = 0.5 * (maxs + mins)
+
+        # Add a small offset between min/max point and the edge of the
+        # plot:
+        deltas = (maxs - mins) / 12.0
```
Comment (QuLogic) [inline lib/mpl_toolkits/mplot3d/axis3d.py:192]: We're Python 3 now, so no need to `.0`
```suggestion
        deltas = (maxs - mins) / 12
```

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
Thanks @Illviljan! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.
